server/src/main/scala/net/wiringbits/actions/SendVerifyEmailAction.scala

```diff
@@ -0,0 +1,68 @@
+package net.wiringbits.actions
+
+import net.wiringbits.apis.models.EmailRequest
+import net.wiringbits.apis.{EmailApi}
+import net.wiringbits.config.{UserTokensConfig, WebAppConfig}
+import net.wiringbits.repositories.UsersRepository
+import net.wiringbits.util.{EmailMessage, TokenGenerator, TokensHelper}
+import net.wiringbits.validations.{ValidateEmailIsRegistered}
+import net.wiringbits.repositories.models.{UserToken, UserTokenType}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import net.wiringbits.api.models.SendVerifyEmail
+import net.wiringbits.repositories.UserTokensRepository
+import java.time.temporal.ChronoUnit
+import java.time.{Clock, Instant}
+import java.util.UUID
+
+class SendVerifyEmailAction @Inject() (
+    usersRepository: UsersRepository,
+    tokenGenerator: TokenGenerator,
+    userTokensConfig: UserTokensConfig,
+    userTokensRepository: UserTokensRepository,
+    webAppConfig: WebAppConfig,
+    emailApi: EmailApi
+)(implicit
+    ec: ExecutionContext,
+    clock: Clock
+) {
+
+  def apply(request: SendVerifyEmail.Request): Future[SendVerifyEmail.Response] = {
+    for {
+      _ <- validations(request)
+
+      userMaybe <- usersRepository.find(request.email)
+      user = userMaybe.getOrElse(throw new RuntimeException(s"User with email ${request.email} wasn't found"))
+
+      token = tokenGenerator.next()
```

Comment:
We already have the logic to generate the token, it would be ideal to reuse it.

Reply:
Is the same logic used in the files "server\src\main\scala\net\wiringbits\actions\CreateUserAction.scala, server\src\main\scala\net\wiringbits\actions\ForgotPasswordAction.scala.

Reply:
`CreateUserAction` logic is almost identical to this action, `ForgotPasswordAction` uses a different token, for now, I'd extract such logic into a method to reuse it, `ForgotPasswordAction` has a slightly different logic which we can abstract later